### PR TITLE
Michael/perf 1699

### PR DESCRIPTION
**Pull request by blonde-mike**
Cleaned Edit URL modal and got endpoint Test button online. Does not yet handle request body or url params, but can make a basic GET request and display the results.

### CODE REVIEW

**Review by summersjc — APPROVED** [review]

**Comment by summersjc** [comment]
Already approved

**Comment by tkmcmaster** [comment]
@blonde-mike Testing this locally it doesn't seem to be working. I'm getting CORS header errors making the requests. on a few test urls.

File: guide/results-viewer-react/src/components/YamlUrls/index.tsx [hunk 1]
```diff
@@ -308,3 +271,126 @@ export function Urls ({ data: { headers, ...data }, ...props }: UrlProps) {
 }
 
 export default Urls;
+
+interface RequestDetailsTabsProps extends HeadersViewProps, ResponseViewProps {}
+
+function RequestDetailsTabs ({ id, headersList, removeHeader, changeHeader, addHeader, responseCode, response }: RequestDetailsTabsProps): JSX.Element {
```
Comment (tkmcmaster) [inline guide/results-viewer-react/src/components/YamlUrls/index.tsx:277]: When creating sub elements, it's a bit cleaner to put them in their own file and create a separate storybook for them.

File: guide/results-viewer-react/src/components/YamlUrls/index.tsx [hunk 2]
```diff
@@ -308,3 +271,126 @@ export function Urls ({ data: { headers, ...data }, ...props }: UrlProps) {
...
+
+interface HeadersViewProps {
+  id: string;
+  headersList: PewPewHeader[];
+  removeHeader: (headerId: string) => void;
+  changeHeader: (headerIndex: number, type: PewPewHeaderStringType, value: string) => void;
+  addHeader: (headerType?: HeaderType) => void;
+}
+
+function HeadersView ({ id, headersList, removeHeader, changeHeader, addHeader }: HeadersViewProps): JSX.Element {
```
Comment (tkmcmaster) [inline guide/results-viewer-react/src/components/YamlUrls/index.tsx:317]: When creating sub elements, it's a bit cleaner to put them in their own file and create a separate storybook for them.

File: guide/results-viewer-react/src/components/YamlUrls/index.tsx [hunk 3]
```diff
@@ -137,65 +143,25 @@ export function Urls ({ data: { headers, ...data }, ...props }: UrlProps) {
...
+      url,
+      headers: headersList.reduce((acc, header) => {
+        acc[header.name] = header.value;
+        return acc;
+      }, {} as Record<string, string>),
+      validateStatus: () => true
+    };
+    try {
+      const response = await axios(request);
+      setLastResponseCode(response.status + " " + response.statusText);
```
Comment (tkmcmaster) [inline guide/results-viewer-react/src/components/YamlUrls/index.tsx:158]: If you're using multiple `useState` hooks together it's more efficient to use a single state object and set them at the same time. Ex. `setLastResponse({ data: response.data, status: response.status + " " + response.statusText })`

File: guide/results-viewer-react/src/components/YamlUrls/index.tsx [hunk 4]
```diff
@@ -137,65 +143,25 @@ export function Urls ({ data: { headers, ...data }, ...props }: UrlProps) {
...
+      setLastResponse(response.data);
+      return response.data;
+    } catch (error) {
+      throw error;
```
Comment (tkmcmaster) [inline guide/results-viewer-react/src/components/YamlUrls/index.tsx:162]: Throwing the error are we displaying the error to the user somehow? Locally I'm having to look at the console to see the CORS errors I was getting.

File: guide/results-viewer-react/src/components/YamlUrls/index.tsx [hunk 5]
```diff
@@ -137,65 +143,25 @@ export function Urls ({ data: { headers, ...data }, ...props }: UrlProps) {
...
-      }
+  async function makeRequest () {
+    const request: AxiosRequestConfig = {
+      method,
+      url,
+      headers: headersList.reduce((acc, header) => {
+        acc[header.name] = header.value;
+        return acc;
+      }, {} as Record<string, string>),
+      validateStatus: () => true
```
Comment (tkmcmaster) [inline guide/results-viewer-react/src/components/YamlUrls/index.tsx:154]: This probably should actually look at the status code. There's other errors like CORS errors that are not being displayed properly

**Review by tkmcmaster — COMMENTED** [review]
@blonde-mike I added some comments to the PR.

File: guide/results-viewer-react/src/components/YamlUrls/index.tsx (hunk 3 above)
Comment (tkmcmaster) [reply guide/results-viewer-react/src/components/YamlUrls/index.tsx:158]: Here's a guide for when to merge vs. keeping them separate.

https://legacy.reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables

File: guide/results-viewer-react/src/components/YamlUrls/index.tsx (hunk 4 above)
Comment (tkmcmaster) [reply guide/results-viewer-react/src/components/YamlUrls/index.tsx:162]: `(error as AxiosError).isAxiosError` will tell you if it's an `AxiosError` type.

**Pull request merged**